clouddriver-cloudfoundry/src/main/java/com/netflix/spinnaker/clouddriver/cloudfoundry/client/Applications.java

```diff
@@ -380,15 +382,14 @@ public void setCurrentDroplet(String appGuid, String dropletGuid) throws CloudFo
       .orElse(ProcessStats.State.DOWN);
   }
 
-  @Nullable
-  public Resource<com.netflix.spinnaker.clouddriver.cloudfoundry.client.model.v2.Application> getLatestServerGroup(String clusterName, String spaceId) {
-    return safelyCall(() -> api.appsLessThanName("name<" + clusterName + "-v9999", "desc", 1))
-      .flatMap(pr -> pr.getResources().stream()
-        .filter(app -> {
-          Names names = Names.parseName(app.getEntity().getName());
-          return clusterName.equals(names.getCluster()) && spaceId.equals(app.getEntity().getSpaceGuid());
-        })
-        .findAny())
-      .orElse(null);
+  public List<Resource<com.netflix.spinnaker.clouddriver.cloudfoundry.client.model.v2.Application>> getTakenSlots(String clusterName, String spaceId) {
+    List<String> filter = asList("name<" + clusterName + "-v9999", "name>=" + clusterName, "space_guid:" + spaceId);
```

Comment:
Wouldn't '>=clusterName-v000' remove the ambiguity?

Reply:
@claymccoy An application called `clusterName` deployed outside of Spinnaker is interpreted by Spinnaker as a v000 sequenced server group, so unfortunately we can't do that.

Reply:
It still doesn't totally remove the ambiguity anyway. An app named `clusterName-zulu-v001` > `clusterName-v000` but needs to be filtered away by cluster name.